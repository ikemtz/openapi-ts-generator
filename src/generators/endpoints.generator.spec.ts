import { EndPointsGenerator } from './endpoints.generator';

describe('EndPointsGenerator', () => {
  it('should generate files', () => {
    const gen = new EndPointsGenerator({
      outputPath: './',
    });
    const response = gen.eliminateDupes({
      paths: [
        { tag: 'lookup', endpoint: 'api/v1/Lookup.{format}' },
        { tag: 'lookup', endpoint: 'api/v1/Lookup.{format}' },
        { tag: 'lookup', endpoint: 'api/v1/Lookup.{format}/GetLookup1' },
        { tag: 'lookup', endpoint: 'api/v1/Lookup.{format}/GetLookup2' },
        { tag: 'lookup_snake_test', endpoint: 'api/v1/Lookup.{format}' },
        { tag: 'lookup_snake_test', endpoint: 'api/v1/Lookup.{format}' },
      ],
    });
    expect(response).toMatchSnapshot();
  });
});
