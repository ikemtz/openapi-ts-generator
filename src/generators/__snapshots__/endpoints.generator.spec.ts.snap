// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndPointsGenerator should generate files 1`] = `
Array [
  Object {
    "endpoint": "api/v1/Lookup.{format}",
    "tag": "lookup",
  },
  Object {
    "endpoint": "api/v1/Lookup.{format}",
    "tag": "lookup2",
  },
  Object {
    "endpoint": "api/v1/Lookup.{format}",
    "tag": "lookupSnakeTest",
  },
  Object {
    "endpoint": "api/v1/Lookup.{format}",
    "tag": "lookupSnakeTest2",
  },
  Object {
    "endpoint": "api/v1/Lookup.{format}/GetLookup1",
    "tag": "lookupGetLookup1",
  },
  Object {
    "endpoint": "api/v1/Lookup.{format}/GetLookup2",
    "tag": "lookupGetLookup2",
  },
]
`;
